### PR TITLE
#22: Fix Logfile and Fix Symbackup

### PR DESCRIPTION
**Pull request by GSaunders92**
Fix Branch error in subscriptions.q. Resolved by splitting if statement on line 61, into two if statements on lines 61 and 101.
 
Fix Symbackup. Issue was no symrecs directory to save backup.  Fixed by adding symrecs directory generation in wdb.q.

Addressed TorQ-Finance-Starter-Pack Issue #22  : 
- Confirm end of day backup is occurring correctly
- Add in generation of backup directory
- Timestamps are replayed in order
- All of the records in the log are successfully appended
- Time spent reading the log
- Distinct syms in log file

### CODE REVIEW

**Comment by ThomasSmyth** [comment]
* You need to include text in the description to say what issue this closes.
* Description could be more detailed.
* Needs assignee, label and project.

File: code/common/subscriptions.q [hunk 1]
```diff
@@ -97,7 +97,12 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
 		.lg.o[`subscribe;"subscription successful"];
 		updatesubscriptions[proc;;instrs]each subtabs];
 			
-		/.lg.o[`DistinctSym;"Total distinct syms: ",string count distinct raze chksym each tables[]];	
+
+/- Count Distinct syms,Display Log file read time, and check log file replayed in order - (branch error fix)
+
+  if[count [details] &  replaylog;
```
Comment (ThomasSmyth) [inline code/common/subscriptions.q]: If you change the order here you can drop the squares.

File: code/common/subscriptions.q [hunk 2]
```diff
@@ -108,8 +113,9 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
                 readTimer-:@[{x};.z.n;{.lg.e[`ReadTimer;"Error calculating timer: ", readEnd]}];
                 /-display result
                 .lg.o[`ReadTimer;"Read Timer total time taken: ", string neg readTimer];
-		/$[1b=min[{all[signum[deltas[exec time from x]]>=0]}each subtabs inter `trade`trade_iex`quote`quote_iex`srcquote`clienttrade];.lg.o[`TimeOrder;"all records have been replayed in timestamp order"];.lg.o[`TimeOrder;" not all records have been replayed in timestamp order"]]];
-			
+    /-Check log file is replayed in time order
+		$[1b=min[{all[signum[deltas[exec time from x]]>=0]}each subtabs inter `trade`trade_iex`quote`quote_iex`srcquote`clienttrade];.lg.o[`TimeOrder;"All records have been replayed in timestamp order"];.lg.o[`TimeOrder;"Not all records have been replayed in timestamp order"]];
```
Comment (ThomasSmyth) [inline code/common/subscriptions.q]: This is way too long, should be a multiline. Also that lambda has squares that can be dropped if you rearrange it.

File: code/common/subscriptions.q [hunk 3]
```diff
@@ -108,8 +113,9 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
...
+    ]
```
Comment (ThomasSmyth) [inline code/common/subscriptions.q]: Semi colon.

File: code/common/subscriptions.q [hunk 4]
```diff
@@ -90,20 +90,18 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
...
 		.lg.o[`subscribe;"subscription successful"];
 		updatesubscriptions[proc;;instrs]each subtabs];
 			
 
-/- Count Distinct syms,Display Log file read time, and check log file replayed in order - (branch error fix)
+/- Count Distinct syms,Display Log file read time, and check log file replayed in order - (branch error fix from prev if statement)
 
-  if[count [details] &  replaylog;
-    /- Count distinct syms
+  if[count [details] & replaylog;
```
Comment (GSaunders92) [inline code/common/subscriptions.q]: Deliberately left in this order left to match order of previous if statements.

Comment (ThomasSmyth) [reply code/common/subscriptions.q]: Turn this caterpillar code into a beautiful butterfly :butterfly: also you need to write `this closes #22` in the description to automate it.

**Pull request merged**